.github/copilot-instructions.md

```diff
@@ -0,0 +1,254 @@
+# GitHub Copilot Instructions for Application Insights JavaScript SDK
+
+## Project Overview
+This is the **Microsoft Application Insights JavaScript SDK** - a browser-based telemetry library for monitoring web applications. The SDK tracks page views, user interactions, performance metrics, exceptions, and custom events.
+
+## Architecture & Structure
+
+### Monorepo Structure
+- **AISKU/**: Main Application Insights SDK package
+- **AISKULight/**: Lightweight version of the SDK
+- **shared/**: Core shared libraries (AppInsightsCore, AppInsightsCommon, 1ds-core-js)
+- **extensions/**: Plugin-based extensions (analytics, dependencies, React, Angular, etc.)
+- **channels/**: Data transmission channels (online, offline, tee)
+- **tools/**: Build and development tools
+- **examples/**: Sample implementations
+
+### Key Technologies
+- **TypeScript/JavaScript**: Primary languages (ES5 target for browser compatibility)
+- **Rush**: Monorepo management tool
+- **Rollup**: Module bundler
+- **Grunt**: Task runner
+- **Dynamic Proto**: Dynamic prototype pattern for performance
+
+## Code Style & Patterns
+
+### TypeScript/JavaScript Conventions
+- Use **ES5-compatible** syntax for browser support
+- Prefer `function` declarations over arrow functions for better IE compatibility
+- Use `var` instead of `let/const` in some contexts for ES5 compatibility
+- Always use semicolons
+- Use 4-space indentation
+- Maximum line length: 140 characters
+
```

Comment:
Need a section (or sections) about the following probably around coding patterns
- Each package should be side-effect free
- All code should be tree-shakable
- Any globals should be lazily initialized to enable tree-shaking using the ILazyValue interface or some simular pattern to determine the difference between the value has not yet been check / assigned vs the value resulting value is null / undefined.
-